fairseq/model_parallel/models/pipeline_parallel_transformer/model.py

```diff
@@ -87,7 +117,10 @@ def forward(self, src_tokens, src_lengths, prev_output_tokens):
         if self.training:
             input_lst = [src_tokens, src_lengths, prev_output_tokens]
             input = tuple(i.to(self.devices[0], non_blocking=True) for i in input_lst)
-            return self.model(input)
+            if TORCH_PIPE:
+                return self.model(input).local_value()
```

Comment:
What does `.local_value()` do?

Reply:
PyTorch Pipe returns an [RRef](https://pytorch.org/docs/stable/rpc.html#rref) pointing to the output of the pipeline. This was introduced in the API to support cross host pipelines in the future. For now, the pipe implementation is only a single host so we can retrieve the value using `local_value` method of the RRef since it will always be local. However, in the future the output of the pipeline could be on a different host and we probably need methods like `to_here` on the RRef to retrieve the remote value.